flutter_mobx/example/lib/todos/todos.dart

```diff
@@ -0,0 +1,185 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_mobx/flutter_mobx.dart';
+import 'package:mobx/mobx.dart';
+part 'todos.g.dart';
+
+class Todo = TodoBase with _$Todo;
+
+abstract class TodoBase implements Store {
+  TodoBase(this.description);
+
+  @observable
+  String description = '';
+
+  @observable
+  bool done = false;
+}
+
+enum VisibilityFilter { all, pending, completed }
+
+class TodoList = TodoListBase with _$TodoList;
+
+abstract class TodoListBase implements Store {
+  @observable
+  ObservableList<Todo> todos = ObservableList<Todo>();
+
+  @observable
+  VisibilityFilter filter = VisibilityFilter.all;
+
+  @computed
+  ObservableList<Todo> get pendingTodos =>
+      ObservableList.of(todos.where((todo) => todo.done != true));
+
+  @computed
+  ObservableList<Todo> get completedTodos =>
+      ObservableList.of(todos.where((todo) => todo.done == true));
+
+  @computed
+  bool get hasCompletedTodos => completedTodos.isNotEmpty;
+
+  @computed
+  bool get hasPendingTodos => pendingTodos.isNotEmpty;
+
+  @computed
+  ObservableList<Todo> get visibleTodos {
+    switch (filter) {
+      case VisibilityFilter.pending:
+        return pendingTodos;
+      case VisibilityFilter.completed:
+        return completedTodos;
+      default:
+        return todos;
+    }
+  }
+
+  @observable
+  String currentDescription = '';
+
+  @action
+  void addTodo(String description) {
+    final todo = Todo(description);
+    todos.add(todo);
+    currentDescription = '';
+  }
+
+  @action
+  void removeTodo(Todo todo) {
+    todos.removeWhere((x) => x == todo);
+  }
+
+  // ignore: use_setters_to_change_properties
```

Comment:
Just disable this lint rule completely, goes against the spirit of actions.

Reply:
Sorry, are you suggesting to disable the rule or NOT to disable? 🤔 😄

Reply:
Disable it completely from the project (and others too)

Reply:
Also I added the action to indicate the semantic operation of changing a filter or setting a  description for the new todo. Unfortunately its a one-liner and failing the lint. I rather not have a stray observable change outside of an action. This is also a design philosophy in MobX and it guards it with a configurable option called `enforceActions`.

Reply:
Remove the ignore comments for the removed rule